simulation/executor/simulate.go

```diff
@@ -161,9 +161,13 @@ func cursedInitializationLogic(
 		ChainID:         config.ChainID,
 		Height:          int64(config.InitialBlockHeight),
 		Time:            genesisTimestamp,
-		ProposerAddress: validators.randomProposer(r),
+		ProposerAddress: validators.randomProposer(r).Address(),
+		AppHash:         res.AppHash,
 	}
 
+	// must set version in order to generate hashes
+	initialHeader.Version.Block = 11
```

Comment:
Do we need to change this hard coded value for every upgrade happening? Or is this hard coded versioning independent?

Reply:
No, this is the block version. For our usecase, we can literally just set it to {anything}. 

Tendermint very infrequently updates this.

It isn't really related to our Osmosis app version.

Reply:
oh wow, didn't realize the block version just happens to 11 in Tendermint, like Osmosis right now: https://github.com/tendermint/tendermint/blob/main/version/version.go#L24

Reply:
Yeah basically when I was trying to figure out the MVP, the block header version was one of the items that has to be set in order to generate a hash. The error message said it was expecting 11, so I simply set it to 11 haha. I guess this will just have to get manually changed whenever TM changes this.